src/rules/authorizedApiGatewaysV2Routes/index.ts

```diff
@@ -0,0 +1,38 @@
+import { Route } from '@aws-sdk/client-apigatewayv2';
+import compact from 'lodash/compact';
+import { fetchAllApiGatewayV2Routes } from '../../aws-sdk-helpers';
+import { SlsMentorLevel } from '../../constants/level';
+import { Category, Rule } from '../../types';
+
+const isAuthenticated = (route: Route): boolean => {
+  const hasAuthorizer =
+    route.AuthorizationType !== undefined && route.AuthorizationType !== 'NONE';
+
+  return hasAuthorizer;
+};
+
+const run: Rule['run'] = async resourceArns => {
+  const apiGatewaysV2Routes = await fetchAllApiGatewayV2Routes(resourceArns);
+  const results = compact(
```

Comment:
il sert à quoi le compact ?

Reply:
It removes all the falsy values from an array (eg. undefined, null...). It allows me to remove checks to see if values are undefined.